deepspeed/inference/v2/modules/implementations/linear/blas_fp_linear.py

```diff
@@ -70,14 +71,15 @@ def __init__(self, config: DSLinearConfig, implementation_config: Dict[str, Any]
                                    dtype=config.output_dtype,
                                    device=get_accelerator().current_device())
 
-    def transform_param(self, param: torch.Tensor) -> torch.Tensor:
+    def transform_param(self, param: torch.Tensor) -> InferenceParameter:
         """
         Converts param to same data type as input and output.
 
         Parameters:
             param (torch.Tensor): Weight or bias tensor.
         """
-        return param.to(self._config.input_dtype)
+        param = param.to(self._config.output_dtype)
```

Comment:
Checking why we need to do self._config.output_dtype instead of self._config.input_dtype?